### ISSUE TITLE
Slow test cases on pypy3.9-7.3.15 on Ubuntu 22.04 

### ISSUE THREAD

**Issue by ecerulm** [urllib3/urllib3#3323]
### Subject

Currently the urllib3's test suite on pypy3.9-7.3.14 and pypy3.9-7.3.15 runs unbearably slow, takes more that 1h45m. 

This is the reason why #3308 pypy3.9-7.3.13 is currently pinned in the GitHub Actions CI. 

The issue has been reported upstream to pypy https://github.com/pypy/pypy/issues/4877 and in coverage https://github.com/nedbat/coveragepy/issues/1738. But after some investigatio it's clear the issue is at the pypy side since the running the test without coverage is still unbearably slow. 

The purpose of this issue is to keep some tracking at the urllib3 level even when the solution (IMHO) should come from pypy.

### Environment

Ubuntu 22.04 (on VirtualBox)
pypy3.9-7.3.14 (from pyenv)

### Steps to Reproduce

```shell
#zsh
pyenv install pypy3.9-7.3.14
path=($(pyenv prefix pypy3.9-7.3.14)/bin $path)
nox -s test-pypy3.9 # takes hours
```

It willl take more that 1h45m to get to 80%, each test is slower that the previous one so it's possible that it will take several hours to complete. 

Memory comsumption until 80%  seems stable .  

### Expected Behavior

The test suite should run under 3 minutes like it does when pypy3.9-7.3.13 is used. 

```shell
#zsh
pyenv install pypy3.9-7.3.13
path=($(pyenv prefix pypy3.9-7.3.13)/bin $path)
nox -s test-pypy3.9 # takes 3 in my environment
```

### Actual Behavior

It takes 1h45m to get to 80% test completion 

Each test takes longer that the previous.  After 1h45m each test can take several minutes to complete. 

If I run any of those tests that take several minutes to complete in isolation on the same environment it finishes instantaneously. 

 

# Related

https://github.com/urllib3/urllib3/pull/3309
#3308

**Comment by mattip** [comment]
We just released pypy3.10-v7.3.17 with some improvements in performance, including packaging OpenSSL 3.3.1. Is it any faster in your CI?

**Comment by ecerulm** [comment]
The CI uses  7.3.16  , see https://github.com/urllib3/urllib3/pull/3452

```
 Successfully set up PyPy 7.3.16 with Python (3.10.14)
```

 As @mattip realized in https://github.com/pypy/pypy/issues/4877#issuecomment-2296066242 , my PR  https://github.com/urllib3/urllib3/pull/3358 fixed the slowness in pypy by closing some threads created by tests (I think it was 10 threads total). So the problem is gone from urllib3 perspective. 
 
 So this issue is closed by #3358 and #3452

**Comment by mattip** [comment]
I would like to see if PyPy3.10, v7.3.17, released last week, is any faster. Could you trigger a build with `check-latest: true` in the [setup-python action](https://github.com/urllib3/urllib3/actions/runs/10672599391/job/29580360861#step:3:5) to force an update the image? Once the image updates (if indeed that is the reason CI is not picking up the new version), you could remove it.
